File: Dsa-day28.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string maxGood = "";
        
        for (int i = 0; i <= (int)num.size() - 3; i++) {
            string sub = num.substr(i, 3);
            if (sub[0] == sub[1] && sub[1] == sub[2]) {
                if (maxGood == "" || sub > maxGood) {
                    maxGood = sub;
                }
            }
        }
        
        return maxGood;
    }
};

File: Dsa-day28.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        max_good = ""
        for i in range(len(num) - 2):
            if num[i] == num[i + 1] == num[i + 2]:
                sub = num[i:i + 3]
                if max_good == "" or sub > max_good:
                    max_good = sub
        return max_good
